### PR TITLE
2.0 cursor collections

### PR DESCRIPTION
**Pull request by Smitsel**

### CODE REVIEW

File: src/Mollie/API/Resources/ResourceFactory.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /**
+     * Create resource object from Api result
+     *
+     * @param object $apiResult
+     * @param BaseResource $resource
+     *
+     * @return BaseResource
+     */
+    public static function createFromApiResult($apiResult, BaseResource $resource)
```
Comment (Daanvm) [inline src/Mollie/API/Resources/ResourceFactory.php]: Can you add a simple unit test for this method?

File: src/Mollie/API/Resources/CursorCollection.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Mollie\Api\Resources;
+
+use Mollie\Api\MollieApiClient;
+
+abstract class CursorCollection extends BaseCollection
```
Comment (Daanvm) [inline src/Mollie/API/Resources/CursorCollection.php]: The refunds collection should also extend from this class.

File: src/Mollie/API/Resources/ResourceFactory.php (hunk 1 above)
Comment (Smitsel) [reply src/Mollie/API/Resources/ResourceFactory.php]: Done.

File: src/Mollie/API/Resources/CursorCollection.php (hunk 2 above)
Comment (Smitsel) [reply src/Mollie/API/Resources/CursorCollection.php]: Done!

**Pull request merged**